### PR TITLE
PP-11314 add publishing workflow for release commits

### PR DESCRIPTION
**Pull request by nlsteers**
### WHAT

Simple action to publish the package when the version has been released by Concourse

### CODE REVIEW

File: .github/workflows/publish-package.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    branches:
+      - main
+
+permissions:
+  contents: read
+
+jobs:
+  publish:
+    runs-on: ubuntu-latest
+    if: "contains(github.event.head_commit.message, '[release]')"
```
[the remark this answers is not in the document]
Comment (nlsteers) [reply .github/workflows/publish-package.yml:14]: I think it's a good idea to lock this pipeline to just run on commits made by the concourse worker, will update.

[the remark this answers is not in the document]
Comment (nlsteers) [reply .github/workflows/publish-package.yml:14]: it's tricker to do this than appears as it will always be the case that someone else will be merging the PR, not the concourse user. This means that the head_commit author will be the merging user and not the concourse user.

File: .github/workflows/prevent-merge-if-release-open.yml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              state: 'open'
+            })
+
+            const openRelease = prs.data.find(pr => pr.user.login === 'alphagov-pay-ci-concourse' && pr.state === 'open')
+
+            if (openRelease) {
+              core.setFailed('There is an unmerged release PR, please merge it before merging this PR.')
+            }
```
Comment (katstevens) [inline .github/workflows/prevent-merge-if-release-open.yml:29]: Will this script work on the `ubuntu-latest` GHA runner? I think we need a `setup-node`

File: .github/workflows/run-tests.yml [hunk 3]
```diff
@@ -1,10 +1,8 @@
-name: Run Tests
+name: Run tests
 
 on:
   pull_request:
-  push:
-    branches:
-      - main
```
Comment (katstevens) [inline .github/workflows/run-tests.yml:7]: If we're getting rid of this, I think we need to add in the `needs: tests` on the `publish` job in `publish-package.yml`?

File: .github/workflows/patch-version.yml
Comment (katstevens) [inline .github/workflows/patch-version.yml:1]: Is this file still needed?

Comment (nlsteers) [reply .github/workflows/patch-version.yml:1]: No, Concourse does this now.

File: .github/workflows/run-tests.yml (hunk 3 above)
Comment (nlsteers) [reply .github/workflows/run-tests.yml:7]: added

**Review by katstevens — APPROVED** [review]
We're looking good 👍 I guess you need to merge the other PR first!
